test/compoundAssetProxyTest.ts

```diff
@@ -0,0 +1,138 @@
+      .approve(fixture.position.address, 10e11);
+    await testFixture.usdc
+      .connect(users[3].user)
+      .approve(testFixture.position.address, 10e11);
+  });
+  after(async () => {
+    // revert back to initial state after all tests pass
+    await restoreSnapshot(provider);
+  });
+  describe("deposit", () => {
+    it("deposits correctly", async () => {
+      await fixture.position
+        .connect(users[0].user)
+        .deposit(users[0].address, 2e6);
+      const balance = await fixture.position.balanceOf(users[0].address);
+      // Allows a 0.01% conversion error
+      expect(balance).to.be.at.least(subError(ethers.BigNumber.from(2e6)));
+    });
+    it("fails to deposit amount greater than available", async () => {
+      const tx = fixture.position
+        .connect(users[1].user)
+        .deposit(users[1].address, 10e12);
+      await expect(tx).to.be.reverted;
+    });
+    it("reverts with compound mint failure", async () => {
+      const tx = testFixture.position
+        .connect(users[3].user)
+        .deposit(users[3].address, 10);
+      await expect(tx).to.be.reverted;
+      // todo: not reverting with the right string, so reverting somewhere else
+    });
+  });
+  describe("withdraw", () => {
+    it("withdraws correctly", async () => {
+      const shareBalance = await fixture.position.balanceOf(users[0].address);
+      await fixture.position
+        .connect(users[0].user)
+        .withdraw(users[0].address, shareBalance, 0);
+      expect(await fixture.position.balanceOf(users[0].address)).to.equal(0);
+    });
+    it("fails to withdraw more shares than in balance", async () => {
+      // withdraw 10 shares from user with balance 0
+      const tx = fixture.position
+        .connect(users[2].user)
+        .withdraw(users[2].address, 10, 0);
+      await expect(tx).to.be.reverted;
+    });
+    it("reverts with compound redeem failure", async () => {
+      const tx = testFixture.position
+        .connect(users[3].user)
+        .withdraw(users[3].address, 10, 0);
+      await expect(tx).to.be.reverted;
+      // todo: not reverting with the right string, so reverting somewhere else
+    });
+  });
+  describe("rewards", () => {
+    it("collects rewards", async () => {
+      // starting balance should be 0
+      expect(await fixture.comp.balanceOf(users[0].address)).to.equal(0);
+      // collect the rewards
+      await fixture.position
+        .connect(users[0].user)
+        .collectRewards(users[0].address);
+      // after collection balance should be nonzero
+      const balance = await (
+        await fixture.comp.balanceOf(users[0].address)
+      ).toNumber();
+      expect(balance).to.be.greaterThan(0);
+    });
+    it("fails for unauthorized user", async () => {
+      const tx = fixture.position
+        .connect(users[1].user)
+        .collectRewards(users[0].address);
+      await expect(tx).to.be.revertedWith("Sender not Authorized");
+    });
+  });
+});
```

Comment:
I like to have the test case for the `withdrawUnderlying` to verify the formula and the `prefundDeposit()` as well.

Reply:
Any update here ?

Reply:
Added a test to use withdrawUnderlying along with the updated math in the `_underlying` function